include/itkMaxPhaseCorrelationOptimizer.hxx

```diff
@@ -75,22 +77,23 @@ MaxPhaseCorrelationOptimizer<TRegistrationMethod>
     throw err;
     }
 
-  const typename ImageType::IndexType
-              index = m_MaxCalculator->GetIndexOfMaximum();
-  const typename ImageType::SizeType
-              size = input->GetLargestPossibleRegion().GetSize();
-  const typename ImageType::SpacingType
-              spacing = input->GetSpacing();
+  const typename ImageType::IndexType index = m_MaxCalculator->GetIndexOfMaximum();
+  const typename ImageType::SizeType size = input->GetLargestPossibleRegion().GetSize();
+  const typename ImageType::SpacingType spacing = input->GetSpacing();
+  const typename ImageType::PointType fixedOrigin = fixed->GetOrigin();
+  const typename ImageType::PointType movingOrigin = moving->GetOrigin();
 
   for( unsigned int i = 0; i < ImageDimension; ++i )
     {
-    if ( index[i] > Math::floor( size[i] / 2.0 ) )
+    OffsetScalarType directOffset = (movingOrigin[i] - fixedOrigin[i]) - 1 * spacing[i] * index[i];
+    OffsetScalarType wrapMIOffset = (movingOrigin[i] - fixedOrigin[i]) - 1 * spacing[i] * (index[i] - IndexValueType(size[i]));
```

Comment:
`MI` means "max index"? Please sio (spell it out ;-) )

Reply:
I meant it as a Middle_of_Image, but what I was actually trying it to make it align with the line above, to make it easy to stop the difference. I now renamed the variable to `mirrorOffset`.